tests/test_work_unit.py

```diff
@@ -284,6 +284,72 @@ def test_save_and_load_fits(self):
             # We succeed if overwrite=True
             work.to_fits(file_path, overwrite=True)
 
+    def test_save_and_load_fits_shard(self):
+        with tempfile.TemporaryDirectory() as dir_name:
+            file_path = os.path.join(dir_name, "test_workunit.fits")
+            self.assertFalse(Path(file_path).is_file())
+
+            # Unable to load non-existent file.
+            self.assertRaises(ValueError, WorkUnit.from_fits_shard, "test_workunit.fits", dir_name)
+
+            # Write out the existing WorkUnit with a different per-image wcs for all the entries.
+            # work = WorkUnit(self.im_stack, self.config, None, self.diff_wcs)
+            work = WorkUnit(im_stack=self.im_stack, config=self.config, wcs=None, per_image_wcs=self.diff_wcs)
+            work.to_fits_shard("test_workunit.fits", dir_name)
+            self.assertTrue(Path(file_path).is_file())
+
+            # Read in the file and check that the values agree.
+            work2 = WorkUnit.from_fits_shard(filename="test_workunit.fits", directory=dir_name)
+            self.assertEqual(work2.im_stack.img_count(), self.num_images)
+            self.assertIsNone(work2.wcs)
+            self.assertFalse(work2.has_common_wcs())
+            for i in range(self.num_images):
+                li = work2.im_stack.get_single_image(i)
+                self.assertEqual(li.get_width(), self.width)
+                self.assertEqual(li.get_height(), self.height)
+                self.assertEqual(li.get_obstime(), 59000.0 + (2 * i + 1))
+
+                # Check the three image layers match.
+                sci1 = li.get_science()
+                var1 = li.get_variance()
+                msk1 = li.get_mask()
+
+                li_org = self.im_stack.get_single_image(i)
+                sci2 = li_org.get_science()
+                var2 = li_org.get_variance()
+                msk2 = li_org.get_mask()
+
+                for y in range(self.height):
```

Comment:
You can use the `RawImage.l2_allclose()` function here. It will be a lot faster and handle NaNs.

```
self.assertTrue(sci1.l2_allclose(sci2, 1e-3))
```